src/System.Private.ServiceModel/tools/test/SelfHostWcfService/WSRequestReplyService.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.Threading;
+
+namespace WcfService
+{
+    [ServiceBehavior(InstanceContextMode = InstanceContextMode.Single)]
+    public class WSRequestReplyService : IWSRequestReplyService
+    {
+        private string ContentToReplace = "ContentToReplace";
+        private string ResponseReplaceThisContent = "ResponseReplaceThisContent";
+        private string ReplacedContent = "ReplacedContent";
+        private string RemoteEndpointMessagePropertyFailure = "RemoteEndpointMessageProperty did not contain the address of this machine.";
+
+        static List<string> log = new List<string>();
+        static int seed = DateTime.Now.Millisecond;
```

Comment:
Same suggestions about 'const' or 'static readonly' strings and considering making log thread safe.